### DISCUSSION TITLE
Programmatic way to open wallet extension on the browser directly without having to wait for the user to click on the button on the connect modal

### DISCUSSION THREAD

**Discussion by 9oelM** [rainbow-me/rainbowkit#1486] [Q&A] [answered]
Hi all, I wanna talk about this modal

![image](https://github.com/rainbow-me/rainbowkit/assets/22465806/c1dc28fe-9cd2-4c61-92a3-0b0faf6889c1)

Upon clicking 'send message', this function is called and will open the wallet extension on the browser:
https://github.com/rainbow-me/rainbowkit/blob/1a0da3f246b01132507b5308e0a6a04eeaf34581/packages/rainbowkit/src/components/SignIn/SignIn.tsx#L55

What I wanna do is to programmatically call this function without having to wait for the user to click on the button in our app verifying your account is a must, so they have to click on the button anyway to be able to use the app. But I had a brief look at the code, and i don't think it's possible to achieve that unless i just copy and paste the entire signIn function to my codebase and somehow make it work right?

I also tried something like

```ts
const interval = window.setInterval(() => {
      const rbkModalRoot = document.querySelector(`[aria-labelledby="rk_connect_title"]`)
      if (rbkModalRoot) {
        const rbkSendAuthMessageButton = rbkModalRoot.querySelector(`[data-testid="rk-auth-message-button"]`)
        if (rbkSendAuthMessageButton) {
          // doesn't work
          (rbkSendAuthMessageButton as HTMLElement).click()
          setTimeout(() => {
            window.clearInterval(interval)
          })
        }
      }
    }, 50)
```

but the click isn't working when it's deployed (works on localhost...) so idk what's going on. has anyone faced a similar problem?

FYI, im not talking about https://www.rainbowkit.com/docs/modal-hooks. I'm already using the hook to open the connect modal. Im talking about the next step which is to open the wallet extension on the browser to prompt the user to sign the message to verify themselves. I want to do this programmatically, which means i want to achieve the same effect of clicking the 'send message' button somehow programmatically.

**Comment by DanielSinclair** [comment, marked as answer]
This is not possible today, but improving this SIWE experience with single-click sign-in is something we would like to work on soon. Do you have any other feature requests for SIWE that we should take into consideration?

**Reply by 9oelM** [reply]
nope that's it for now for me.

btw i just got it working with something like

```ts
useEffect(() => {
    const cleanup = () => {
      if (elemCheckInterval.current) {
        window.clearInterval(elemCheckInterval.current)
      }
    }
    if (!connectModalOpen) {
      return cleanup
    }
    const interval = window.setInterval(() => {
      const rbkModalRoot = document.querySelector(`[aria-labelledby="rk_connect_title"]`)
      if (rbkModalRoot) {
        const rbkSendAuthMessageButton = rbkModalRoot.querySelector(`[data-testid="rk-auth-message-button"]`) as HTMLElement | undefined
        if (rbkSendAuthMessageButton) {
          rbkSendAuthMessageButton.click()
        }
      }
    }, 50)
    elemCheckInterval.current = interval

    return cleanup
  }, [connectModalOpen])
```

I think `setTimeout` in my previous code was the problem. I know it just a hack, but it works for now. Thanks for the prompt reply.
